### ISSUE TITLE
Hide settings panel differently

### ISSUE THREAD

**Issue by borondics** [biolab/orange3#6937]
This might belong to `orange-canvas-core` or `orange-widget-base`. Sorry if I am adding the issue in the wrong place.

**What's your use case?**

Many widgets have a functionality to hide the settings panel on the left to be able to maximize the plot area. This is now a multi-step process, because the settings panel simply disappears and this makes the widget window smaller, which needs to be resized by the user.
<img width="1074" alt="image" src="https://github.com/user-attachments/assets/aa19031e-61c3-44d2-a223-10dfd9b028e6">

Actually, I just realized that this behaviour (at least on MacOS 14.6.1) is not consistent when the window is maximized or not (see video below).

https://github.com/user-attachments/assets/44011e6d-8f02-4c5f-97cb-430fb6df43c0

**What's your proposed solution?**

Would it be an easy thing to change the behaviour of the "hide settings panel" functionality so that the widget window size remains the same and the plot area fills up the space previously taken by the settings panel?
<img width="1096" alt="image" src="https://github.com/user-attachments/assets/f5370544-bdd7-4cb7-984c-57657c63dc1a">

In other words, it would be great to have the full screen behavior in both maximized and non-maximized cases.

**Are there any alternative solutions?**
It is possible to do this manually in multiple steps.

**Comment by markotoplak** [comment]
Actually what Orange currently does for non-maximized windows was carefully thought about and additional steps were taken for it to be this way. The goal was not to change the aspect ration of the visualization.

What you propose changes aspect ratios (which we can not avoid if maximized, because it is maximized).

**Comment by borondics** [comment]
Yeah, I see your point of the aspect ratio of the plot and indeed, my suggestion would change that. 

However, what I would consider in this case is what happens currently after you hide the panel? Do you leave the widget window untouched or you change the window size? Then I have two questions: first, if you don't change the window, why did you hide the panel, second, if you do change it, how do you make sure the aspect ratio stays the same as before... 

One could think about calculating the aspect ratio of the plot before the change and then using this to enlarge the window automatically?

**Comment by markotoplak** [comment]
What happens now is that only the control area disappears and that is it. This nicely supports uses in setting the plot however they want and then they can hide the control area and still expect that the plot did not change.
 
This was designed as a way for the users to have more active (smaller) windows on a screen where they can show/hide control panels at will, which should not affect visualizations or even their positions. 

It was not designed as a way to see the biggest possible plots; that is what maximizing a widget is for.

**Comment by markotoplak** [comment]
So, basically, we have two different reasons for hiding control area:
1. Enabling more widgets on the screen at the same time.
2. Having as big graphs as possible.

The current behaviour aims to optimize (1). This, I think is a better default.
